ansible/roles/hana_secondary_actions/tasks/main.yml

```diff
@@ -0,0 +1,183 @@
+---
+- name: Prepare secondary Action - Dump cluster status
+  command: crm status
+  become: true
+  register: crm_initial
+  changed_when: false
+
+- name: Prepare secondary Action - Assert this node is not master
+  assert:
+    that:
+      - crm_initial.stdout is not search('\* Masters:\s*\[\s*{{ inventory_hostname }}\s*\]')
+    fail_msg: "{{ inventory_hostname }} is MASTER – expected replica"
+    success_msg: "{{ inventory_hostname }} is in replica mode"
+
+- name: Prepare secondary Action - Wait cluster idle (pre-action)
+  command: cs_wait_for_idle --sleep 5
+  become: true
+  register: idle_pre
+  retries: "{{ cs_wait_timeout // 5 }}"
+  delay: 5
+  until: idle_pre.rc == 0
+
+# SBD-related stuff
+- name: Prepare secondary Action - Gather SBD-delay parameters
+  become: true
+  shell: "{{ item.cmd }}"
+  args:
+    warn: false
+  register: sbddelay
+  changed_when: false
+  failed_when: false
+  loop:
+    - name: corosync_token
+      cmd: "corosync-cmapctl | awk -F ' = ' '/totem.token/ {print int($2/1000)}'"
+    - name: corosync_consensus
+      cmd: "corosync-cmapctl | awk -F ' = ' '/totem.consensus/ {print int($2/1000)}'"
+    - name: pcmk_delay_max
+      cmd: "crm resource param stonith-sbd show pcmk_delay_max | sed 's/[^0-9]*//g'"
+    - name: sbd_watchdog_timeout
+      cmd: "grep -oP '(?<=^SBD_WATCHDOG_TIMEOUT=)[[:digit:]]+' /etc/sysconfig/sbd"
+    - name: sbd_delay_start
+      cmd: "grep -oP '(?<=^SBD_DELAY_START=)([[:digit:]]+|yes|no)+' /etc/sysconfig/sbd"
+  when: action == 'crash'
```

Comment:
why this one does not have condition on EC2 ?

Reply:
>> Are we poring an OSADO bug? :-)

Yes... :p in this initial port, I change nothing (or at least try to - there are a few changes here and there as a result of me trying to dig into called functions and dependencies)